homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,335 @@
+            return
+
+        if CONF_NAME in repository:
+            self.name = repository[CONF_NAME]
+        else:
+            self.name = repo.name
+
+        self.update = Throttle(interval)(self._update)
+
+        self.stargazers = None
+        self.topics = None
+        self.branches = None
+        self.open_issues = None
+        self.open_pull_requests = None
+        self.last_commit = None
+        self.latest_release = None
+        self.available = False
+
+    def _update(self):
+        """Update GitHub Sensor."""
+        try:
+            repo = self._github_obj.get_repo(self.repository_path)
+
+            self.stargazers = repo.stargazers_count
+
+            self.topics = json.dumps(repo.get_topics())
+
+            branches = []
+            for branch in repo.get_branches():
+                branches.append(branch.name)
+            self.branches = json.dumps(branches)
+
+            open_issues = []
+            for issue in repo.get_issues(state='open', sort='created'):
+                if issue.pull_request is None:
+                    labels = []
+                    for label in issue.labels:
+                        labels.append(label.name)
+                    open_issues.append({
+                        "number": issue.number,
+                        "title": issue.title,
+                        "url": issue.html_url,
+                        "body": issue.body,
+                        "labels": labels,
+                        "assignee": issue.assignee,
+                        "milestone": issue.milestone,
+                        "comments": issue.comments,
+                        "user": {
+                            "login": issue.user.login,
+                            "name": issue.user.name,
+                            "avatar_url": issue.user.avatar_url,
+                            "url": issue.user.html_url
+                        }
+                    })
+            self.open_issues = json.dumps(open_issues)
+
+            open_pull_requests = []
+            for _pr in repo.get_pulls(state='open', sort='created'):
+                labels = []
+                for label in _pr.labels:
+                    labels.append(label.name)
+                open_pull_requests.append({
+                    "number": _pr.number,
+                    "title": _pr.title,
+                    "url": _pr.html_url,
+                    "body": _pr.body,
+                    "labels": labels,
+                    "assignee": _pr.assignee,
+                    "milestone": _pr.milestone,
+                    "comments": _pr.comments,
+                    "user": {
+                        "login": _pr.user.login,
+                        "name": _pr.user.name,
+                        "avatar_url": _pr.user.avatar_url,
+                        "url": _pr.user.html_url
+                    }
+                })
+            self.open_pull_requests = json.dumps(open_pull_requests)
```

Comment:
This is way too much information, all this stuff gets stored in the recorder at every change. Maybe a count would be enough?

Reply:
How about a count and storing just the latest PR? That way you can send a notification etc. with the PR details (Same as issues)